components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts

```diff
@@ -26,21 +26,31 @@ export class ProjectsFilterDropdownComponent implements OnChanges {
   @Input() filterVisible = false;
 
   @Output() onSelection = new EventEmitter<ProjectsFilterOption[]>();
+  @Output() onOptionChange = new EventEmitter<ProjectsFilterOption[]>();
 
   editableOptions: ProjectsFilterOption[] = [];
+  filteredOptions: ProjectsFilterOption[] = [];
+
+  filterValue = '';
 
   optionsEdited = false;
 
   dropdownActive = false;
 
+  handleFilterKeyUp() {
+    this.filteredOptions = this.editableOptions.filter(option =>
+      option.label.toLowerCase().indexOf(this.filterValue.toLowerCase()) > -1
+    );
+  }
+
   ngOnChanges(changes) {
     if (changes.options && !this.optionsEdited) {
       this.resetOptions();
     }
   }
```

Comment:
It is difficult to conceptualize when this is going to be hit and why it is appropriate. (I confirm it does work as is, though!)
However, what do you think about the following to make the code easier to follow?
1. Get rid of `ngOnChanges`.
2. Make the following tweaks to `optionsEdited` and `resetOptions`:
<img width="537" alt="image" src="https://user-images.githubusercontent.com/6817500/72859666-80edd100-3c79-11ea-8a21-0d39bc6f9642.png">
The one subtle difference is that `handleLabelClick` is only resetting upon entry, not upon exit, but we there is no need to do it upon exit.

Reply:
I like this approach better - thanks for the suggestion

Reply:
@SEAjamieD I believe you missed step 1 here (get rid of ngOnChanges)--that's important, too.

Reply:
@msorens @SEAjamieD the original intention of this particular `ngOnChanges` was to ensure the `editableOptions` copy was updated if/whenever the `options @Input` value changed:

https://github.com/chef/automate/blob/29c87ea67edf45025cb1ce6efece6c768c1742c0/components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts#L29-L36

Removing `ngOnChanges` will cause copies of the `options @Input` to be stale should the host component ever change the value of this input.